sgx-hasher/src/lib.rs

```diff
@@ -79,6 +80,52 @@ impl Hasher {
     }
 }
 
+pub fn sign(
+    author: sig::Author,
+    contents: sig::Contents,
+    key: rsa::Rsa<pkey::Private>,
+) -> std::io::Result<sig::Signature> {
+    // Generates signature on Signature author and contents
+    let rsa_key = pkey::PKey::from_rsa(key.clone())?;
+    let md = hash::MessageDigest::sha256();
+    let mut signer = sign::Signer::new(md, &rsa_key)?;
+    signer.update(author.as_ref())?;
+    signer.update(contents.as_ref())?;
+    let signature = signer.sign_to_vec()?;
+
+    // Generates q1, q2 values for RSA signature verification
+    let s = bn::BigNum::from_slice(&signature)?;
+    let m = key.n();
+    let e = u32::try_from(key.e()).expect("Could not convert BigNum to u32");
```

Comment:
`expect()` is not correct. I'm also worried that upstream rust-openssl isn't going to merge my `TryFrom` patches. We should probably write our own converter.